### ISSUE TITLE
Loading pretrained weights after extending a model

### ISSUE THREAD

**Issue by PickHub** [pytorch/fairseq#1289]
I was wondering if there was a way to leverage a pretrained model after adding another component to the model.

Specifically I'm extending [MASS](https://github.com/microsoft/MASS). When fine-tuning the untouched model I _fairseq-train_ with the _load-from-pretrained-model_ flag. Now I want to add a second encoder with different inputs, change to a multi-task loss and use the same decoder. Can I initialize encoder/decoder with the pretrained weights? And if yes, how? Thanks for your help!

**Comment by myleott** [comment]
Sure, you can modify your model to do this. Take a look at the stories generation model to see how they did this: https://github.com/pytorch/fairseq/blob/master/fairseq/models/fconv_self_att.py#L109-L127
